dbf-reader/src/main/java/org/jamel/dbf/structure/DbfHeader.java

```diff
@@ -109,18 +107,17 @@ public short getRecordLength() {
         return recordLength;
     }
 
-    int getFieldIndex(String fieldName) {
+    public int getFieldIndex(String fieldName) {
         if (fieldIndexesByNames == null) {
             initFieldIndexesByNames();
         }
-        if (!fieldIndexesByNames.containsKey(fieldName)) {
-            throw new DbfException(format("Field \"%s\" does not exist", fieldName));
-        }
-        return fieldIndexesByNames.get(fieldName);
+        Integer index = fieldIndexesByNames.get(fieldName);
+        return index == null ? -1 : index;
     }
 
     private void initFieldIndexesByNames() {
-        fieldIndexesByNames = new HashMap<>();
+        int capacity = (int) (fields.size() / 0.75 + 1);
+        fieldIndexesByNames = new HashMap<>(capacity);
```

Comment:
As far as I understand, the Hashmap is going to be rehashed when the number of elements exceeds the initial capacity multiplied by load factor. So if I create
```
new HashMap<>(10);
```
it will be rehashed when I add 8 elements to it.
From the javadoc:

> If the initial capacity is greater than the maximum number of entries divided by the load factor, no rehash operations will ever occur.